src/in/shick/diode/markdown/Markdown.java

```diff
@@ -79,7 +79,7 @@
     static final RunAutomaton autoLinkUrlAutomaton = new RunAutomaton(new RegExp("((https?|ftp):([^'\"> \t\r\n])+)", RegExp.NONE).toAutomaton());
 //    static final Pattern autoLinkEmail = Pattern.compile("<([-.\\w]+\\@[-a-z0-9]+(\\.[-a-z0-9]+)*\\.[a-z]+)>");
 
-    static final RunAutomaton subredditAutomaton = new RunAutomaton(new RegExp("/[rR]/[a-zA-Z0-9]+/?", RegExp.NONE).toAutomaton());
+    static final RunAutomaton subredditAutomaton = new RunAutomaton(new RegExp("[rR]/[a-zA-Z0-9]+/?", RegExp.NONE).toAutomaton());
```

Comment:
This new regex is a little too broad.  Looks like it would hotlink /r/fire out of 'water/fire type'.

Instead of just removing the leading slash from the pattern, you could replace it with ([^a-zA-Z0-9]|^) which seems like a better compromise to me.

Unfortunately that will slightly complicate the MarkdownURL code since you'll have to deal with the possible presence of a leading character in the match that's neither / nor r.